covidfaq/rerank/predict.py

```diff
@@ -0,0 +1,61 @@
+from functools import lru_cache
+
+# import boto3
+import torch
+from transformers import BertModel, BertTokenizer
+
+from covidfaq.rerank.train_retriever import (
+    BertEncoder,
+    Retriver,
+    RetriverTrainer,
+)
+
+# from zipfile import ZipFile
+
+
+@lru_cache()
+def load_model():
+    ## Berts
+    model_str = "bert-base-uncased"
+    tokenizer = BertTokenizer.from_pretrained(model_str)
+    bert_question = BertModel.from_pretrained(model_str)
+    bert_paragraph = BertModel.from_pretrained(model_str)
+    max_question_len_global = 30
+    max_paragraph_len_global = 512
+
+    encoder_question = BertEncoder(bert_question, max_question_len_global)
+    encoder_paragarph = BertEncoder(bert_paragraph, max_paragraph_len_global)
+    ret = Retriver(encoder_question, encoder_paragarph, tokenizer)
+
+    model = RetriverTrainer(ret)
+
+    # s3 = boto3.client("s3")
+    # s3.download_file(
+    #     "coviddata.dialoguecorp.com",
+    #     "mirko/bert_rerank_model__for_testing.zip",
+    #     "covidfaq/rerank/model.zip",
+    # )
+
+    # with ZipFile("covidfaq/rerank/model.zip", "r") as zip:
+    #     zip.extractall()
```

Comment:
This is the code to load and extract the model right?

Reply:
yes